File: src/136.single-number.js
/*
 * @lc app=leetcode id=136 lang=javascript
 *
 * [136] Single Number
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    var answer = 0;

    nums.forEach(val => {
      answer ^= val;
    });

    return answer;
};
// @lc code=end

File: Top Interview Questions/src/136.single-number.java
/*
 * @lc app=leetcode id=136 lang=java
 *
 * [136] Single Number
 */

// @lc code=start
class Solution {
  public int singleNumber(int[] nums) {
    int temp = 0;

    for (int each : nums) {
      temp ^= each;
    }

    return temp;
  }
}
// @lc code=end

